pkg/application/registryConfigUseCase.go

```diff
@@ -0,0 +1,270 @@
+
+	for _, diff := range diffs {
+		switch diff.NeededAction {
+		case domain.ConfigActionSet:
+			entry := &ecosystem.DoguConfigEntry{
+				Key:   common.DoguConfigKey{DoguName: doguName, Key: diff.Key.Key},
+				Value: common.DoguConfigValue(diff.Expected.Value),
+			}
+			entriesToSet = append(entriesToSet, entry)
+		case domain.ConfigActionRemove:
+			keysToDelete = append(keysToDelete, common.DoguConfigKey{DoguName: doguName, Key: diff.Key.Key})
+		case domain.ConfigActionNone:
+			continue
+		default:
+			errs = append(errs, doguUnknownConfigActionError(diff.NeededAction, diff.Key.Key, doguName))
+		}
+	}
+
+	if len(entriesToSet) > 0 {
+		errs = append(errs, useCase.doguConfigRepository.SaveAll(ctx, entriesToSet))
+	}
+	if len(keysToDelete) > 0 {
+		errs = append(errs, useCase.doguConfigRepository.DeleteAllByKeys(ctx, keysToDelete))
+	}
+
+	return errors.Join(errs...)
+}
+
+// Values of sensitiveConfig from existing Dogus are already encrypted???
+func (useCase *EcosystemRegistryUseCase) applySensitiveDoguConfigDiffs(ctx context.Context, doguName common.SimpleDoguName, diffs domain.SensitiveDoguConfigDiffs) error {
+	var errs []error
+
+	var entriesToSet []*ecosystem.SensitiveDoguConfigEntry
+	var keysToDelete []common.SensitiveDoguConfigKey
+	var entriesToEncrypt []*ecosystem.SensitiveDoguConfigEntry
+
+	for _, diff := range diffs {
+		switch diff.NeededAction {
+		case domain.ConfigActionSet:
+			entry := getSensitiveDoguConfigEntry(doguName, diff)
+			entriesToSet = append(entriesToSet, entry)
+		case domain.ConfigActionSetToEncrypt:
+			entry := getSensitiveDoguConfigEntry(doguName, diff)
+			entriesToEncrypt = append(entriesToEncrypt, entry)
+		case domain.ConfigActionRemove:
+			keysToDelete = append(keysToDelete, common.SensitiveDoguConfigKey{DoguConfigKey: common.DoguConfigKey{DoguName: doguName, Key: diff.Key.Key}})
+		case domain.ConfigActionNone:
+			continue
+		default:
+			errs = append(errs, doguUnknownConfigActionError(diff.NeededAction, diff.Key.Key, doguName))
+		}
+	}
+
+	if len(entriesToSet) > 0 {
+		errs = append(errs, useCase.doguSensitiveConfigRepository.SaveAll(ctx, entriesToSet))
+	}
+	if len(keysToDelete) > 0 {
+		errs = append(errs, useCase.doguSensitiveConfigRepository.DeleteAllByKeys(ctx, keysToDelete))
+	}
+	if len(entriesToEncrypt) > 0 {
+		errs = append(errs, useCase.doguSensitiveConfigRepository.SaveAllForNotInstalledDogu(ctx, doguName, entriesToEncrypt))
+	}
+
+	return errors.Join(errs...)
+}
+
+func getSensitiveDoguConfigEntry(doguName common.SimpleDoguName, diff domain.SensitiveDoguConfigEntryDiff) *ecosystem.SensitiveDoguConfigEntry {
+	return &ecosystem.SensitiveDoguConfigEntry{
+		Key:   common.SensitiveDoguConfigKey{DoguConfigKey: common.DoguConfigKey{DoguName: doguName, Key: diff.Key.Key}},
+		Value: common.EncryptedDoguConfigValue(diff.Expected.Value),
+	}
+}
+
+func getDoguConfigEntry(doguName common.SimpleDoguName, key, value string) *ecosystem.DoguConfigEntry {
```

Comment:
🚫 **[checkstyle]** <unused> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>func `getDoguConfigEntry` is unused